java/src/main/java/ai/rapids/cudf/Scalar.java

```diff
@@ -373,6 +373,90 @@ public static Scalar listFromColumnView(ColumnView list) {
     return new Scalar(DType.LIST, makeListScalar(list.getNativeView(), true));
   }
 
+  /**
+   * Creates a null scalar of struct type.
+   *
+   * @param elementTypes data types of children in the struct
+   * @return a null scalar of struct type
+   */
+  public static Scalar structFromNull(HostColumnVector.DataType... elementTypes) {
+    ColumnVector[] children = new ColumnVector[elementTypes.length];
+    long[] childHandles = new long[elementTypes.length];
+    try {
+      for (int i = 0; i < elementTypes.length; i++) {
+        // Build column vector having single null value rather than empty column vector,
+        // because struct scalar requires row count of children columns == 1.
+        children[i] = buildNullColumnVector(elementTypes[i]);
+        childHandles[i] = children[i].getNativeView();
+      }
+      return new Scalar(DType.STRUCT, makeStructScalar(childHandles, false));
+    } finally {
+      IllegalStateException closeException = null;
```

Comment:
This won't do what we want.  If an exception is thrown from the `try` block then this `finally` block will still clobber that exception if any `close()` call fails.  It needs something like this:
```java
  RuntimeException error = null;
  try {
    ...
  } catch (RuntimeException e) {
    error = e;
  } catch (Exception e) {
    error = new RuntimeException(e);
  } finally {
    for (ColumnVector child : children) {
      if (child != null) {
        try {
          child.close();
        } catch (Exception e) {
          if (error != null) {
            error.addSuppressed(e);
          } else {
            error = new RuntimeException(e);
          }
        }
      }
    }
    if (error != null) {
      throw error;
    }
  }
```

Reply:
Fixed. Thanks for the demo code!